loopout.go

```diff
@@ -719,11 +721,21 @@ func (s *loopOutSwap) payInvoiceAsync(ctx context.Context,
 	paymentSuccess := err == nil &&
 		paymentStatus.State == lnrpc.Payment_SUCCEEDED
 
-	if err := s.swapKit.server.ReportRoutingResult(
-		ctx, s.swapInfo().SwapHash, s.swapInvoicePaymentAddr, pluginType,
-		paymentSuccess, int32(attempts), dt.Milliseconds(),
-	); err != nil {
-		s.log.Warnf("Failed to report routing result: %v", err)
+	if reportPluginResult {
+		// If the plugin couldn't be acquired then override the reported
+		// plugin type to RoutingPluginNone.
+		reportType := pluginType
+		if routingPlugin == nil {
+			reportType = RoutingPluginNone
+		}
```

Comment:
Don't need this? The enum will default to `RoutingPluginNone` ?

Reply:
The reason for this is because routingPlugin can be nil if the acquire failed (even though plugin type wasn't none).

Reply:
Right, thought that `pluginType` would always be the zero value, got it!